### PR TITLE
RUBY-1133 Implement MaxStaleness spec

### PR DESCRIPTION
**Pull request by estolfo**

### CODE REVIEW

**Comment by estolfo** [comment]
@ajdavis @evanchooly If you would be able to do a code review, I'd really appreciate it. Thanks in advance!

File: lib/mongo/server_selector/selectable.rb [hunk 1]
```diff
@@ -212,9 +220,48 @@ def match_tag_sets(candidates)
...
+        if !@tag_sets.all? { |set| set.empty? } && !tags_allowed?
+          raise Error::InvalidServerPreference.new(Error::InvalidServerPreference::NO_TAG_SUPPORT)
+        elsif @max_staleness && !max_staleness_allowed?
+          raise Error::InvalidServerPreference.new(Error::InvalidServerPreference::NO_MAX_STALENESS_SUPPORT)
+        end
+      end
+
+      def validate_max_staleness_support!(server)
+        return unless @max_staleness
+        if @max_staleness && !server.features.collation_enabled?
```
Comment (ajdavis) [inline lib/mongo/server_selector/selectable.rb]: Rename "collation_enabled" to something more generic, or make it a separate property? I understand why this works but it looks surprising.

**Comment by ajdavis** [comment]
Looks correct to me with my limited knowledge - very nice.

**Comment by estolfo** [comment]
thanks @ajdavis !

**Pull request merged**